src/main/java/com/datapipe/jenkins/vault/credentials/VaultTokenCredentialBinding.java

```diff
@@ -29,25 +29,31 @@
 
     private final static String DEFAULT_VAULT_ADDR_VARIABLE_NAME = "VAULT_ADDR";
     private final static String DEFAULT_VAULT_TOKEN_VARIABLE_NAME = "VAULT_TOKEN";
+    private final static String DEFAULT_VAULT_NAMESPACE_VARIABLE_NAME = "VAULT_NAMESPACE";
 
     @NonNull
     private final String addrVariable;
     private final String tokenVariable;
+    private final String namespaceVariable;
     private final String vaultAddr;
+    private final String vaultNamespace;
 
     /**
      *
      * @param addrVariable if {@code null}, {@value DEFAULT_VAULT_ADDR_VARIABLE_NAME} will be used.
      * @param tokenVariable if {@code null}, {@value DEFAULT_VAULT_TOKEN_VARIABLE_NAME} will be used.
+     * @param namespaceVariable if {@code null}, {@value DEFAULT_VAULT_NAMESPACE_VARIABLE_NAME} will be used.
      * @param credentialsId credential identifier
      * @param vaultAddr vault address
      */
     @DataBoundConstructor
-    public VaultTokenCredentialBinding(@Nullable String addrVariable, @Nullable String tokenVariable, String credentialsId, String vaultAddr) {
+    public VaultTokenCredentialBinding(@Nullable String addrVariable, @Nullable String tokenVariable, @Nullable String namespaceVariable, String credentialsId, String vaultAddr, String vaultNamespace) {
```

Comment:
Please move your additions to `@DataBoundSetter` as the namespace is not mandatory and this breaks binary compatibility :sweat:
```suggestion
    public VaultTokenCredentialBinding(@Nullable String addrVariable, @Nullable String tokenVariable, String credentialsId, String vaultAddr) {
```